tests/datatypes/test_tuple.py

```diff
@@ -272,3 +272,22 @@ class BinaryTupleOperationTests(BinaryOperationTestCase, TranspileTestCase):
 
 class InplaceTupleOperationTests(InplaceOperationTestCase, TranspileTestCase):
     data_type = 'tuple'
+
+    substitutions = {
+        # all possible outputs
+        "{1.2, 3, True}": [
+            "{1.2, True, 3}"
+        ],
+        "{3, 1.2, True}": [
+            "{1.2, True, 3}"
+        ],
+        "{3, True, 1.2}": [
+            "{1.2, True, 3}"
+        ],
+        "{True, 3, 1.2}": [
+            "{1.2, True, 3}"
+        ],
+        "{True, 1.2, 3}": [
+            "{1.2, True, 3}"
+        ],
```

Comment:
Closer, but we're still not there. 

The format is *output*: *input list*. We want a *single* output, keyed to *all* possible inputs - the idea is that *any* ordered input will be mapped to a single, canonical ordering.

Reply:
Is my formatting all correct? Am I supposed to remove the quotation marks?

Also, should I be adding a `substitutions` feature into `class UnaryTupleOperationTests` and `class BinaryTupleOperationTests`?

Reply:
The quotation marks are required; the substitutions are being done on the string output.

You *don't* need to add this to UnaryTupleOperationTests and class BinaryTupleOperationTests. However, you do need to  make the change to the test that is actually raising the error. The test failure is with BuiltinSetFunctionTests; that's the only class that needs substitutions. You're adding a substitution definition to `datatypes/test_tuple.py`; the error is being raised by `builtins/test_tuple.py`

You also need to revert the indentation change you've made to `tests/builtins/test_tuple.py`.